packages/web/src/repositories/pool/pool-repository-impl.ts

```diff
@@ -683,13 +684,32 @@ export class PoolRepositoryImpl implements PoolRepository {
     tokenA: TokenModel,
     tokenB: TokenModel,
     feeTier: SwapFeeTierType,
-    startPrice: string,
+    startPriceStr: string,
     caller: string,
   ) {
     const tokenAPath = tokenA.wrappedPath || tokenA.path;
     const tokenBPath = tokenB.wrappedPath || tokenB.path;
     const fee = `${SwapFeeTierInfoMap[feeTier].fee}`;
-    const startPriceSqrt = tickToSqrtPriceX96(priceToTick(Number(startPrice)));
+    const startPrice = BigNumber(startPriceStr).toNumber();
+
+    /**
+     * If the token path pairs are out of order, adjust the price and token order.
+     */
+    const isOrdered = isOrderedTokenPaths(tokenAPath, tokenBPath);
+    if (!isOrdered) {
+      const reverseStartPriceSqrt =
+        startPrice !== 0 ? tickToSqrtPriceX96(priceToTick(1 / startPrice)) : 0;
+
+      return {
+        caller,
+        send: "",
+        pkg_path: POOL_PATH,
+        func: "CreatePool",
+        args: [tokenBPath, tokenAPath, fee, reverseStartPriceSqrt.toString()],
```

Comment:
shouldn't it be `tokenAPath`, `tokenBPath` ??

Reply:
Shouldn't the token order be reversed?

Reply:
Oh yeah, you're right.
I was confused by variable name